nemo/collections/nlp/models/token_classification/punctuation_capitalization_model.py

```diff
@@ -365,92 +368,163 @@ def _setup_infer_dataloader(
+        acc_prob = acc_prob[number_of_probs_to_move:]
+        return pred, acc_prob
+
+    @staticmethod
+    def update_accumulated_probabilities(acc_prob, update):
+        acc_prob *= update[: acc_prob.shape[0]]
+        acc_prob = np.concatenate([acc_prob, update[acc_prob.shape[0] :]], axis=0)
+        return acc_prob
+
+    @staticmethod
+    def remove_margins(tensor, margin_size, keep_left, keep_right):
+        if not keep_left:
+            tensor = tensor[margin_size + 1 :]  # remove left margin and CLS token
+        if not keep_right:
+            tensor = tensor[: tensor.shape[0] - margin_size - 1]  # remove right margin and SEP token
+        return tensor
+
+    def apply_punct_capit_predictions(self, query, punct_preds, capit_preds):
+        query = query.strip().split()
+        assert len(query) == len(
+            punct_preds
+        ), f"len(query)={len(query)} len(punct_preds)={len(punct_preds)}, query[:30]={query[:30]}"
+        assert len(query) == len(
+            capit_preds
+        ), f"len(query)={len(query)} len(capit_preds)={len(capit_preds)}, query[:30]={query[:30]}"
+        punct_ids_to_labels = {v: k for k, v in self._cfg.punct_label_ids.items()}
+        capit_ids_to_labels = {v: k for k, v in self._cfg.capit_label_ids.items()}
+        query_with_punct_and_capit = ''
+        for j, word in enumerate(query):
+            punct_label = punct_ids_to_labels[punct_preds[j]]
+            capit_label = capit_ids_to_labels[capit_preds[j]]
+
+            if capit_label != self._cfg.dataset.pad_label:
+                word = word.capitalize()
+            query_with_punct_and_capit += word
+            if punct_label != self._cfg.dataset.pad_label:
+                query_with_punct_and_capit += punct_label
+            query_with_punct_and_capit += ' '
+        return query_with_punct_and_capit[:-1]
+
+    def _transform_logit_to_prob_and_remove_margins_and_extract_word_probs(
+        self, punct_logits, capit_logits, subtokens_mask, start_word_ids, margin, is_first, is_last, query_ids,
+    ):
+        new_start_word_ids = list(start_word_ids)
+        subtokens_mask = subtokens_mask > 0.5
+        b_punct_probs, b_capit_probs = [], []
+        for i, (first, last, q_i, pl, cl, stm) in enumerate(
+            zip(is_first, is_last, query_ids, punct_logits, capit_logits, subtokens_mask)
+        ):
+            if not first:
+                new_start_word_ids[i] += torch.count_nonzero(stm[: margin + 1]).numpy()  # + 1 is for [CLS] token
+            stm = self.remove_margins(stm, margin, keep_left=first, keep_right=last)
+            for b_probs, logits in [(b_punct_probs, pl), (b_capit_probs, cl)]:
+                p = torch.nn.functional.softmax(
+                    self.remove_margins(logits, margin, keep_left=first, keep_right=last)[stm], dim=-1,
+                )
+                b_probs.append(p.detach().cpu().numpy())
+        return b_punct_probs, b_capit_probs, new_start_word_ids
```

Comment:
Please add docstrings and types to the newly added methods. How about using leading underscore in the names of all these functions?